lightly/api/api_workflow_predictions.py

```diff
@@ -0,0 +1,169 @@
+from concurrent.futures import ThreadPoolExecutor
+from typing import Dict, List, Union, Optional, Tuple
+
+import tqdm
+
+from lightly.api.prediction_singletons import (
+    PredictionSingletonClassificationRepr,
+    PredictionSingletonRepr,
+)
+from lightly.api.utils import retry
+from lightly.openapi_generated.swagger_client import (
+    PredictionTaskSchema,
+    SamplePartialMode,
+    PredictionSingletonClassification,
+    PredictionSingletonObjectDetection,
+    PredictionSingletonInstanceSegmentation,
+    PredictionSingletonKeypointDetection,
+)
+
+
+class _PredictionsMixin:
+    def create_or_update_prediction_task_schema(
+        self,
+        schema: PredictionTaskSchema,
+        prediction_version_id: int = -1,
+    ) -> None:
+        """Creates or updates the prediction task schema
+
+        Args:
+            schema:
+                The prediction task schema.
+            prediction_version_id:
+                And id to distinguish different predictions for the same sample.
```

Comment:
Its just how we pull together the schemas with the predictions. Aka it could be that suddenly there are more categoryIds with newer predictions.

I would rewrite it slightly;
```
 A numerical id (e.g timestamp) to distinguish different predictions of different model versions. Use the same id if you don't require versioning or if you wish to overwrite the previous schema. Optional, set to -1 by default.
```

Reply:
One thing I don't understand: How do we know which schema belong to a prediction?